core/utils/useragent.js

```diff
@@ -12,104 +12,106 @@
  */
 'use strict';
 
-/**
- * @name Blockly.utils.userAgent
- * @namespace
- */
-goog.provide('Blockly.utils.userAgent');
+goog.module('Blockly.utils.userAgent');
+goog.module.declareLegacyNamespace();
 
-goog.require('Blockly.utils.global');
+const globalUtils = goog.require('Blockly.utils.global');
 
 
-/** @const {boolean} */
-Blockly.utils.userAgent.IE;
+/** @type {boolean} */
+let IE;
+exports.IE = IE;
```

Comment:
Does this approach work? It seems like these are all going to be exported as `undefined` since their values aren't set until later.